tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAwsVersioning.js

```diff
@@ -0,0 +1,548 @@
+                    err => next(err)),
+                next => _awsGetLatestVerId(key, someBody, next),
+                (awsVerId, next) => delAndAssertResult(s3, { bucket,
+                    key, versionId: 'null', resultType: deleteVersion },
+                    err => next(err, awsVerId)),
+                (awsVerId, next) => _awsGetAssertDeleted({ key,
+                    versionId: awsVerId, errorCode: 'NoSuchVersion' }, next),
+            ], done);
+        });
+
+        it('versioning enabled: should delete a specific version in AWS ' +
+        'backend successfully', done => {
+            const key = `somekey-${Date.now()}`;
+            async.waterfall([
+                next => putVersionsToAws(s3, bucket, key, [someBody],
+                    (err, versionIds) => next(err, versionIds[0])),
+                (s3vid, next) => _awsGetLatestVerId(key, someBody,
+                    (err, awsVid) => next(err, s3vid, awsVid)),
+                (s3VerId, awsVerId, next) => delAndAssertResult(s3, { bucket,
+                    key, versionId: s3VerId, resultType: deleteVersion },
+                    err => next(err, awsVerId)),
+                (awsVerId, next) => _awsGetAssertDeleted({ key,
+                    versionId: awsVerId, errorCode: 'NoSuchVersion' }, next),
+            ], done);
+        });
+
+        it('versioning not configured: deleting existing object should ' +
+        'not return version id or x-amz-delete-marker: true but should ' +
+        'create a delete marker in aws ', done => {
+            const key = `somekey-${Date.now()}`;
+            async.waterfall([
+                next => putToAwsBackend(s3, bucket, key, someBody,
+                    err => next(err)),
+                next => delAndAssertResult(s3, { bucket, key,
+                    resultType: nonVersionedDelete }, err => next(err)),
+                next => _getAssertDeleted(s3, { key, errorCode: 'NoSuchKey' },
+                    next),
+                next => _awsGetAssertDeleted({ key, errorCode: 'NoSuchKey' },
+                    next),
+            ], done);
+        });
+
+        it('versioning suspended: should create a delete marker in s3 ' +
+        'and aws successfully when deleting existing object', done => {
+            const key = `somekey-${Date.now()}`;
+            async.waterfall([
+                next => putNullVersionsToAws(s3, bucket, key, [someBody],
+                    err => next(err)),
+                next => delAndAssertResult(s3, { bucket, key, resultType:
+                    newDeleteMarker }, err => next(err)),
+                next => _getAssertDeleted(s3, { key, errorCode: 'NoSuchKey' },
+                    next),
+                next => _awsGetAssertDeleted({ key, errorCode: 'NoSuchKey' },
+                    next),
+            ], done);
+        });
+
+        it('versioning suspended: creating a delete marker should overwrite ' +
```

Comment:
if versioning is suspended on our end, don't you mean delete on our s3 should put a delete marker on aws?